contracts/src/token/erc1155/mod.rs

```diff
@@ -0,0 +1,1975 @@
+//! Implementation of the ERC-1155 token standard.
+use alloc::{vec, vec::Vec};
+
+use alloy_primitives::{fixed_bytes, Address, FixedBytes, U256};
+use openzeppelin_stylus_proc::interface_id;
+use stylus_sdk::{
+    abi::Bytes,
+    alloy_sol_types::sol,
+    call::{self, Call, MethodError},
+    evm, msg,
+    prelude::{public, sol_storage, AddressVM, SolidityError},
+    storage::TopLevelStorage,
+};
+
+use crate::utils::{
+    introspection::erc165::{Erc165, IErc165},
+    math::storage::SubAssignUnchecked,
+};
+
+mod receiver;
+pub use receiver::IERC1155Receiver;
+
+/// `bytes4(
```

Comment:
Nit: we could also use statically computed `function_selector!` macro from stylus sdk. And for `IErc721Receiver` also. But it is fine as it right now